lib/Macros/Sources/SwiftMacros/DebugDescriptionMacro.swift

```diff
@@ -0,0 +1,287 @@
+
+                var propertyChain: [String] = []
+                if let declRef = expr.as(DeclReferenceExprSyntax.self) {
+                    // A reference to a single property on self.
+                    propertyChain.append(declRef.baseName.text)
+                } else if let memberAccess = expr.as(MemberAccessExprSyntax.self) {
+                    // A chain of properties, possibly starting with an explicit self.
+                    enumerateMembers(memberAccess) { declRef in
+                        propertyChain.append(declRef.baseName.text)
+                    }
+                } else {
+                    // The expression was neither a DeclReference nor a MemberAccess.
+                    let message: ErrorMessage = "unsupported expression; stored properties only"
+                    context.diagnose(node: expr, message: message)
+                    return []
+                }
+
+                // Explicit self are removed before use by LLDB.
+                if propertyChain[0] == "self" {
+                    propertyChain.removeFirst()
+                }
+
+                // Check that the root property is not a computed property on `self`.
+                // Ideally, all properties would be checked, but a macro expansion can
+                // at best only check the properties of the type it's attached to.
+                let rootProperty = propertyChain[0]
+                guard !computedProperties.contains(rootProperty) else {
+                    let message: ErrorMessage = "cannot reference computed properties"
+                    context.diagnose(node: expr, message: message)
+                    return []
+                }
+
+                let propertyPath = propertyChain.joined(separator: ".")
+                summarySegments.append("${var.\(propertyPath)}")
+            @unknown default:
+                return []
+            }
+}
+
+        let summaryString = summarySegments.joined()
+
+        var typeIdentifier: String
+        if let typeParameters = genericParameters(declaration), typeParameters.count > 0 {
+            let typePatterns = Array(repeating: ".+", count: typeParameters.count).joined(separator: ",")
+            // A regex matching that matches the generic type.
+            typeIdentifier = "^\(moduleName)\\.\(typeName)<\(typePatterns)>"
+        } else if declaration.is(ExtensionDeclSyntax.self) {
+            // When attached to an extension, the type may or may not be a generic type.
+            // This regular expression handles both cases.
+            typeIdentifier = "^\(moduleName)\\.\(typeName)(<.+>)?$"
+        } else {
+            typeIdentifier = "\(moduleName).\(typeName)"
+        }
+
+        // Serialize the type summary into a global record, in a custom section, for LLDB to load.
+        let decl: DeclSyntax = """
+        #if os(Linux)
+        @_section(".lldbsummaries")
+        #elseif os(Windows)
+        @_section(".lldbsummaries")
+        #else
```

Comment:
is there an `os(Darwin)` that you could use here? I think most other platforms use ELF, so it would make most sense if the ELF section name is the catch-all.

Reply:
there isn't an `os(Darwin)`, see https://docs.swift.org/swift-book/documentation/the-swift-programming-language/statements/#Compiler-Control-Statements

the best we can do is enumerate the currently known Darwin OSes